contracts/L2/ETHxPoolV5.sol

```diff
@@ -0,0 +1,171 @@
+import { AccessControlUpgradeable } from "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import { SafeERC20, IERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title ETHxPool Contract
+ * @author Stader Labs
+ * @notice This contract is responsible for the swap of ETHx; the user must deposit ETH in order to receive ETHx in
+ * return.
+ */
+interface AggregatorV3Interface {
+    function latestRoundData()
+        external
+        view
+        returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound);
+}
+
+contract ETHxPoolV5 is AccessControlUpgradeable, ReentrancyGuardUpgradeable {
+    using SafeERC20 for IERC20;
+
+    /// @notice Role hash of BRIDGER
+    bytes32 public constant BRIDGER_ROLE = keccak256("BRIDGER_ROLE");
+    /// @notice Base rate of ETHx/ETH
+    uint256 public constant ETHX_BASE_RATE = 1e18;
+    /// @notice Address of ETHx token
+    IERC20 public ETHx;
+    /// @notice Basis points for fees
+    uint256 public feeBps;
+    /// @notice Total fees earned in swap
+    uint256 public feeEarnedInETH;
+    /// @notice Address of the ETHx/ETH oracle
+    address public ethxOracle;
+
+    /// @notice Emitted when swap occured successfully
+    event SwapOccurred(address indexed user, uint256 ETHxAmount, uint256 fee, string referralId);
+    /// @notice Emitted when accumulated fees is withdrawn
+    event FeesWithdrawn(uint256 feeEarnedInETH);
+    /// @notice Emitted when deposited ETH is withdrawn
+    event AssetsMovedForBridging(uint256 ethBalanceMinusFees);
+    /// @notice Emitted when basis fee is updated
+    event FeeBpsSet(uint256 feeBps);
+    /// @notice Emitted when oracle address is updated
+    event OracleSet(address indexed oracle);
+
+    /// @dev Thrown when input is zero address
+    error ZeroAddress();
+    /// @dev Thrown when input is invalid amount
+    error InvalidAmount();
+    /// @dev Thrown when input is invalid basis fee
+    error InvalidBps();
+    /// @dev Thrown when transfer is failed
+    error TransferFailed();
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @dev Initialize the contract
+    /// @param _admin The admin address
+    /// @param _bridger The bridger address
+    /// @param _ethx The ETHx token address
+    /// @param _feeBps The fee basis points
+    /// @param _ethxOracle The ethxOracle address
+    function initialize(
+        address _admin,
+        address _bridger,
+        address _ethx,
+        uint256 _feeBps,
+        address _ethxOracle
+    )
+        public
+        initializer
+    {
+        _checkNonZeroAddress(_ethx);
+        _checkNonZeroAddress(_ethxOracle);
+
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+        _setupRole(BRIDGER_ROLE, _bridger);
```

Comment:
Please provide bridger role to admin as well.

Reply:
[7a5d7ef](https://github.com/stader-labs/ethx_oft/pull/8/commits/7a5d7ef77afbebf5f1685638c9da8b5b5154fb70)